dashboard/app/models/user.rb

```diff
@@ -1736,7 +1737,7 @@ def migrated?
 
   def sponsored?
     if migrated?
-      authentication_options.empty?
+      authentication_options.empty? && encrypted_password.blank?
```

Comment:
See [this Slack discussion](https://codedotorg.slack.com/archives/C6B838SB0/p1529513018000813):  There are a few types of accounts that may have no authentication options, and not all of them are sponsored.

- Sponsored accounts (word and picture passwords) have no authentication options _and_ no encrypted password.
- Legacy "manual" accounts sign in with a username and password - they require no authentication options.
- Parent-owned accounts are word and picture accounts where the user has added a username, password, and parent email.  They require no authentication options.

We're choosing not to store an authentication option for username+password logins because we don't ever want a user to have more than one username.

So for now this check (no authoptions and no password) seems to be the correct interpretation of "sponsored."  This will get more complicated later when we deal with Google Classroom student accounts, etc - but not worrying about that yet.